src/test/java/ru/training/at/hw2/BaseJdiTestingIndexPageTest.java

```diff
@@ -0,0 +1,72 @@
+package ru.training.at.hw2;
+
+import java.util.concurrent.TimeUnit;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.asserts.SoftAssert;
+import ru.training.at.hw2.data.JdiTestingIndexPageData;
+
+public class BaseJdiTestingIndexPageTest extends SeleniumTestsSetup {
+    protected WebDriver driver;
+
+    protected SoftAssert assertAccumulator;
+
+    @BeforeClass(alwaysRun = true)
+    public void setUp() {
+        driver = getDriver();
+        driver.manage().window().maximize();
+        driver.manage().timeouts().implicitlyWait(1000, TimeUnit.MILLISECONDS);
+        assertAccumulator = new SoftAssert();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void afterClassTearDown() {
+        if (driver != null) {
+            driver.quit();
+        }
+    }
+
+    protected void openTestSiteByURL() {
+        // 1. Open test site by URL
+        driver.navigate().to(JdiTestingIndexPageData.PAGE_URL);
+        assertAccumulator.assertEquals(driver.getCurrentUrl(), JdiTestingIndexPageData.PAGE_URL);
+    }
+
+    protected void assertBrowserTitle() {
+        // 2. Assert Browser title
+        assertAccumulator.assertEquals(driver.getTitle(), JdiTestingIndexPageData.PAGE_TITLE);
+    }
+
+    protected void performLogin() {
+        // 3. Perform login
+        driver.findElement(LoginFormLocators.DROPDOWN).click();
+        driver.findElement(LoginFormLocators.LOGIN_INPUT).sendKeys(JdiTestingIndexPageData.LOGIN);
+        driver.findElement(LoginFormLocators.PASSWORD_INPUT).sendKeys(JdiTestingIndexPageData.PASSWORD);
+        driver.findElement(LoginFormLocators.LOGIN_BUTTON).click();
+
+        assertAccumulator.assertTrue(driver.findElement(LoginFormLocators.LOGOUT_BUTTON).isDisplayed());
+    }
```

Comment:
I suggest splitting actions and assertions into different methods 
Parameterize these methods so no data is hardcoded inside the method (like username, password, SoftAssert instance, expected values, and so on)

\+ same for other methods